pagopa/io-app
# chore: [IOBP-282,IOBP-287,IOBP-303] Restyling transaction summary screen

## ⚠️ This PR depends on https://github.com/pagopa/io-app-design-system/pull/95

## Short description
This PR introduces a restyling of the `TransactionSummaryScreen` and removes the deprecated feature flag `NEW_TRANSACTION_SUMMARY_ENABLED`.

## List of changes proposed in this pull request
- Upgrading `@pagopa/io-app-design-system` library to the version `^1.12.0` that supports new icons;
- Removed the unused feature flag to show/hide the new transaction screen
- Replaced the old and deprecated `TransactionSummaryScreen` with the new one
- Adapted the design of the new transaction summary screen following the design system guidelines;
- Restyled `TransactionSummaryRow` component with the use of the new List items provided by design system
- Upgraded `io-app-design-system` library with the new prop `loading` to the `ButtonSolid` component

## How to test
Navigate into a transaction summary page after uploading a qrcode, then you should see the new summary screen that is compliant to [the following figma](https://www.figma.com/file/KY9BpP4LaSSd18fuYMwxYK/Paga-un-avviso-pagoPA?type=design&node-id=11-930&mode=design&t=H5KE7JSSAS4vzqkX-4)

## Preview
<img src="https://github.com/pagopa/io-app/assets/34343582/5227fa09-19ae-4213-945c-3c0f63158c90" width="300" />
<img src="https://github.com/pagopa/io-app/assets/34343582/3e29605e-64ae-4785-ab91-8f1db4911d90" width="300" />
